ResearchKit/Common/ORKWorkoutUtilities.swift

```diff
@@ -0,0 +1,379 @@
+    
+    /**
+     Initializer that uses an `identifier` string
+     
+     @param identifier    The identifier to convert
+     @return              An `HKWorkoutActivityType`. Default = `.other`
+     */
+    public init(identifier: String) {
+        guard let idx = ORKWorkoutActivityTypeIdentifiers.index(of: identifier),
+            let type = HKWorkoutActivityType(rawValue: UInt(idx) + 1)
+            else {
+                self = .other
+                return
+        }
+        self = type
+    }
+    
+    /**
+     String identifier for this enum value.
+     */
+    public var identifier: String {
+        let idx = self.rawValue
+        guard idx >= 1 && idx <= UInt(ORKWorkoutActivityTypeIdentifiers.count) else {
+            return "other"
+        }
+        return ORKWorkoutActivityTypeIdentifiers[Int(idx) - 1]
+    }
+}
+
+public extension HKWorkoutEventType {
+    
+    /**
+     Initializer that uses an `identifier` string
+     
+     @param identifier    The identifier to convert
+     @return              An `HKWorkoutEventType` or `nil` if undefined.
+     */
+    public init?(identifier: String) {
+        guard let idx = ORKWorkoutEventTypeIdentifiers.index(of: identifier),
+            let type = HKWorkoutEventType(rawValue: idx + 1) else {
+                return nil
+        }
+        self = type
+    }
+    
+    /**
+     String identifier for this enum value.
+    */
+    public var identifier: String {
+        let idx = self.rawValue
+        guard idx >= 1 && idx <= ORKWorkoutEventTypeIdentifiers.count else {
+            return "other"
+        }
+        return ORKWorkoutEventTypeIdentifiers[idx - 1]
+    }
+}
+
+@available(iOS 10.0, *)
+@available(watchOS 3.0, *)
+public extension HKWorkoutSessionLocationType {
+    
+    /**
+     Initializer that uses an `identifier` string
+     
+     @param identifier    The identifier to convert
+     @return              An `HKWorkoutSessionLocationType`. Default = `.unknown`
+     */
+    public init(identifier: String) {
+        guard let idx = ORKWorkoutSessionLocationTypeIdentifiers.index(of: identifier),
+            let type = HKWorkoutSessionLocationType(rawValue: Int(idx) + 1)
+            else {
+                self = .unknown
+                return
+        }
+        self = type
+    }
+    
+    /**
+     String identifier for this enum value.
+     */
+    public var identifier: String {
+        let idx = self.rawValue
+        guard idx >= 1 && idx <= ORKWorkoutSessionLocationTypeIdentifiers.count else {
+            return ORKWorkoutSessionLocationTypeIdentifiers[0]
+        }
+        return ORKWorkoutSessionLocationTypeIdentifiers[idx - 1]
+    }
+    
+}
+
+// MARK: Identifier map
+
+fileprivate let ORKWorkoutActivityTypeIdentifiers = [   "americanFootball",
+                                                        "archery",
+                                                        "australianFootball",
```

Comment:
wait, they have australian football as its own activity but not tai chi? wtf apple